ranger/ext/widestring.py

```diff
@@ -5,7 +5,32 @@
 from __future__ import (absolute_import, division, print_function)
 
 import sys
-from unicodedata import east_asian_width
+
+try:
+    from wcwidth import wcwidth, wcswidth
+
+    def uwid(string):
+        """Return the width of a string"""
+        if not PY3:
+            string = string.decode('utf-8', 'ignore')
+        return wcswidth(string)
+
+    def utf_char_width(char):
+        """Return the width of a single character"""
+        return wcwidth(char)
+except ImportError:
+    from unicodedata import east_asian_width
+
+    def uwid(string):
+        if not PY3:
+            string = string.decode('utf-8', 'ignore')
+        return sum(utf_char_width(c) for c in string)
```

Comment:
May I suggest,
```
         return sum(max(0, utf_char_width(c)) for c in string)
```

To account for possible return value of -1 for non-printable characters